### PR TITLE
Future doc

### LINKED ISSUE #31728: Documentation for Distributed.Future is just for the constructor
```
help?> Distributed.Future
  Future(pid::Integer=myid())

  Create a Future on process pid. The default pid is the current process.
```

The documentation should be for the type and its behaviour, not (only) for one of its constructors

### PR DESCRIPTION
**Pull request by matbesancon**
Added documentation for `Future` the type, complementing the documentation for one constructor already given. References to relevant other functions
fixes #31728

### CODE REVIEW

**Comment by matbesancon** [comment]
Fixes #31728

File: stdlib/Distributed/src/remotecall.jl [hunk 1]
```diff
@@ -177,9 +185,10 @@ If the argument `Future` is owned by a different node, this call will block to w
 It is recommended to wait for `rr` in a separate task instead
 or to use a local [`Channel`](@ref) as a proxy:
 
-    c = Channel(1)
-    @async put!(c, remotecall_fetch(long_computation, p))
-    isready(c)  # will not block
+    p = 1
```
Comment (fredrikekre) [inline stdlib/Distributed/src/remotecall.jl]: Maybe make this a proper
````
```julia
```
````
while you are at it?

**Review by fredrikekre — APPROVED** [review]

File: stdlib/Distributed/src/remotecall.jl (hunk 1 above)
Comment (matbesancon) [reply stdlib/Distributed/src/remotecall.jl]: @fredrikekre no indentation required in that case right?

Comment (fredrikekre) [reply stdlib/Distributed/src/remotecall.jl]: No.

**Comment by matbesancon** [comment]
no idea why travis would fail on that

**Comment by matbesancon** [comment]
ok the passing & failing CI seems completely random, I would dare saying this is ready for review

**Comment by matbesancon** [comment]
ping @ViralBShah @StefanKarpinski @KristofferC for final review and merging

File: stdlib/Distributed/src/remotecall.jl [hunk 2]
```diff
@@ -232,7 +243,7 @@ function del_clients(pairs::Vector)
     end
 end
 
-any_gc_flag = Condition()
+const any_gc_flag = Condition()
```
Comment (fredrikekre) [inline stdlib/Distributed/src/remotecall.jl]: Docs look fine, but I can not comment on this one. Is there a reason this was non-`const`?

Comment (matbesancon) [reply stdlib/Distributed/src/remotecall.jl]: I can't see why. `Condition` wraps a vector and that's it. There is no other affectation to the `any_gc_flag` in `Distributed`

**Comment by matbesancon** [comment]
bump here

**Comment by matbesancon** [comment]
bump

**Pull request merged**